quickwit-indexing/src/actors/packager.rs

```diff
@@ -205,17 +207,36 @@ fn create_packaged_split(
         .sum();
 
     // Extracts tag values from `_tags` special fields.
-    let mut tags = BTreeSet::default();
     let index_reader = split
         .index
         .reader_builder()
         .reload_policy(ReloadPolicy::Manual)
         .try_into()?;
-    for reader in index_reader.searcher().segment_readers() {
-        let inv_index = reader.inverted_index(tags_field)?;
-        let mut terms_streamer = inv_index.terms().stream()?;
-        while let Some((term_data, _)) = terms_streamer.next() {
-            tags.insert(String::from_utf8_lossy(term_data).to_string());
+    let mut tags = BTreeSet::default();
+    for (tag_field_name, tag_field) in tag_fields_list {
```

Comment:
mmmm, I think `Field` is specific to a segment, not an index. The field resolving needs to be made at the segment level

Reply:
Are you implying segments within the same index can have a different structure or is this taking into account schema change?

Reply:
Ok, here it is safe, let's keep what you did.